DXgovernance/davi-monorepo
# Feat: scheme selection page

# Description

Adds a scheme selection page to the "Create Proposal" process for Governance 1.5 DAOs.

The scheme selection page shows all the relevant capabilities for the scheme, as well as the permissions and the treasury it has.

A user cannot create a proposal in a DAO without a selected scheme.

A user can create a proposal with or without a discussion reference.

Screenshot:
![Screenshot from 2023-03-13 09-42-30](https://user-images.githubusercontent.com/75996796/224708861-49ed779e-423e-4b45-9671-e1be5fd3a86e.png)

Other changes:

1. Subgraph: linked Scheme entity with VotingMachine entity through `@derivedFrom` property
2. Proposal and Treasury pages now accept a `subdao` param to get the information to that address instead of the DAO/Guild address
3. Radio input: basic ones are white and styled
4. Gov 1.5 permissions: now differentiates between all, token and function permissions

**Important note: added `useCreateProposal` and `useVoteOnProposal` placeholder hooks (returns void) to prevent crashing.**

[Figma refence](https://www.figma.com/file/Z2yCtX1NLhdJYYoBiXkdMg/DAVI-v1?node-id=1%3A24032&t=fwxnisVmWeBZ2ISh-1)
Closes #17 

## Type of change



# How Has This Been Tested?

Manual testing. Guilds behavior should be unaffected.

To test this:

1. Go to a DAO page (to get the DAO address, make a query on the subgraph)
6. Create a new proposal, either by going to a discussion and making a proposal from that, by going to "All proposals" and clicking the "create proposal" button or just going to "/create-proposal" URL
7. It should show the /scheme-selection page

In a guild, creating a proposal or going to /scheme-selection page should always led to /create-proposal page

# Checklist:
